### PR TITLE
[Transforms] Merge master 07/11 into transform

### PR DESCRIPTION
**Pull request by yuit**
Merge master into transforms :tada:  There are tests related to `classExpressionWithStaticProperties` that fail. This require this [PR](https://github.com/Microsoft/TypeScript/pull/8631) @rbuckton  ~~should I just accept baselines for now as we already has issue to track this problem?~~

Interesting commit starts from [dd0cd58](https://github.com/Microsoft/TypeScript/commit/a1b626034ebdee814c4e2e8fa96905b0b18cd08b)

TODO:

- [x] Port JSDOC unittests into the transforms

### CODE REVIEW

**Comment by yuit** [comment]
All tests failure related to classExpression which contains capturing static property

**Comment by rbuckton** [comment]
:+1:

File: src/compiler/transformers/ts.ts [hunk 1]
```diff
@@ -3239,14 +3214,14 @@ namespace ts {
         }
 
         function trySubstituteDecoratedClassName(node: Identifier): Expression {
```
Comment (yuit) [inline src/compiler/transformers/ts.ts]: You may want to rename this function since it is not just decorated class name anynmore

**Comment by yuit** [comment]
:+1:

**Pull request merged**